velox/functions/lib/Re2Functions.h

```diff
@@ -114,7 +116,68 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures();
 /// prefix, and suffix patterns. Return the pair {pattern kind, number of '_'
 /// characters} for patterns with wildcard characters only. Return
 /// {kGenericPattern, 0} for generic patterns).
-PatternMetadata determinePatternKind(StringView pattern);
+PatternMetadata determinePatternKind(
+    StringView pattern,
+    std::optional<char> escapeChar);
+
+/// Return the unescaped string for the specified string range, if escape char
+/// is not specified just return the corresponding substring.
+std::string unescape(
```

Comment:
Can we move this function to .cpp file?

Reply:
I left it here to facilitate testing, will move it.